src/main/java/fun/epak/pak/controller/AuthController.java

```diff
@@ -20,8 +26,12 @@ public String getLogin() {
     }
 
     @PostMapping("/register")
-    public RedirectView postNewUser(UserRegistrationRequest userRegistrationRequest) {
-        userService.registerUser(userRegistrationRequest);
+    public RedirectView postNewUser(UserRegistrationRequest userRegistrationRequest,
+                                    @RequestParam("image") MultipartFile multipartFile) throws IOException {
+        User user = userService.registerUser(userRegistrationRequest, multipartFile);
+        String uploadDir = "data/images/profiles/" + user.getId();
```

Comment:
Cała ta logika powinna być w serwisie w `registerUser` a nie w kontrolerze. Kontroler powinien być tak głupi jak to tylko możliwe.